resource/resource_test.go

```diff
@@ -309,6 +309,62 @@ func TestOutE2E(t *testing.T) {
 	}
 }
 
+func TestTargetURL(t *testing.T) {
+	testCases := map[string]struct {
+		atc          string
+		team         string
+		pipeline     string
+		job          string
+		buildN       string
+		instanceVars string
+		want         string
+	}{
+		"all defaults": {
+			want: "https://ci.example.com/teams/devs/pipelines/magritte/jobs/paint/builds/42",
+		},
+		"instanced vars 1": {
+			instanceVars: `{"branch":"stable"}`,
+			want:         "https://ci.example.com/teams/devs/pipelines/magritte/jobs/paint/builds/42?vars=%7B%22branch%22%3A%22stable%22%7D",
+		},
+		"instanced vars 2": {
+			instanceVars: `{"branch":"stable","foo":"bar"}`,
```

Comment:
@aoldershaw according to your comment https://github.com/Pix4D/cogito/pull/26#discussion_r598955336 in the previous PR I have the impression I should nest the JSON ?

Reply:
Sorry, not 100% sure what you mean by nest here, but AFAICT the implementation you have is correct